### PR TITLE
add cached build based of run-cmake and run-vcpkg actions

### PR DESCRIPTION
**Pull request by lukka**

### CODE REVIEW

**Comment by lukka** [comment]
@kaosumaru i meant to create this PR on my repo :) Anyway, as a reference this is how to pipeline would look like using run-cmake and run-vcpkg actions.

**Comment by Kaosumaru** [comment]
Nice, run-cmake & run-vcpkg actions are much better than manually writing bash steps 👍 
I will check & merge it later, thanks!

But I wonder, I understand that you are adding vcpkg as submodule, because you need to have vcpkg binary in workspace to cache it (machines that are running actions have vcpkg preinstalled, but as far as I know there is no way to cache artifacts from it, since they are outside project workspace).

Would it be better to download vcpkg from some version tag (or something) in your action? And then cache binaries.
Then, you wouldn't need to add vcpkg as submodule in project that wants to use run-vcpkg actions.

**Comment by lukka** [comment]
@Kaosumaru thanks for reviewing this PR! About vcpkg, this is the reason of vcpkg being a submodule:
1) any project (and this project as well) needs to use a specific version of vcpkg, i.e. a specific commit id of the repository you use;
2) the best way to store the commit id, is to specify it permanently in the project's Git repository: either using a submodule, or storing the commit id hash in a text file (under source control) would be both viable solutions.

The reason for point 1 and 2 are more thoroughly [explained here](https://github.com/lukka/CppBuildTasks/blob/master/README.md#use-vcpkg-as-a-submodule-of-your-git-repository). Or to just to give two examples: 
- your project may rely on boost 1.68, but vcpkg on the build server is upgraded unexpectedly, and it is using boost 1.72, and now your project does not build anymore on CI because there are breaking changes between the two versions of boost;
- also you want to ensure you use exactly the same vcpkg version, locally and remotely on CI servers;

In short hosting vcpkg on the CI server does not really make sense, as vcpkg's port and its executable are tightly coupled, that is the reason it needs to be re-built for every different commit id.

Let me know if you have more questions!

**Comment by Kaosumaru** [comment]
Solid reasoning, I agree with points 1 & 2.
And it goes without saying, that you should have some depedency versioning in your project. In case of this project, I'm not afraid about breaking changes in SDL2, but in any case, it's really unwise to have no control of versioning of dependencies on your CI machine.

I wanted to propose that you would alternatively allow for setting vcpkg commit SHA as parameter for devs that don't want to use vcpkg as submodule (and I could understand that, because fresh checkout of vcpkg with only SDL2 takes 1GB, and my local vcpkg folder after installing QT & llvm came up to 202 GB) - but from cursory glance on your action repo, it's already done, and you support vcpkgGitCommitId :)

**Comment by lukka** [comment]
@Kaosumaru yes, I can certainly make vcpkg not a submodule, and using an explicit vcpkgGitCommitId parameter instead as you correctly pointed out. Few thoughts before doing so:
 - when you check out your repo, you can choose to not checkout the submodule of vcpkg, so it is not needed to. Only on CI it is needed. Locally you can put it everywhere, as long as you use the correct path its CMake toolchain file;
 - about the size of the vcpkg folder, is it a problem to have such a big subfolder (the vcpkg one) under the same directory of your repository? Note you could get rid of "download", "packages" and especially "buildtrees" to save a ton of space; I think `--clean-after-build` would do so automatically when you run `vcpkg install --clean-after-build @response_file.txt`

Let me know, thanks!
